crypto/sig_abstractions.go

```diff
@@ -127,14 +144,14 @@ type PackedSignatureAlgorithm struct {
 
 var errUnknownSigner = errors.New("could not find stored signer")
 
-func (p *PackedSignatureAlgorithm) getSigner(t AlgorithmType) (Signer, error) {
+func (p *PackedSignatureAlgorithm) getSigner(t AlgorithmType) Signer {
 	switch t {
 	case DilithiumType:
-		return &p.DilithiumSigner, nil
+		return &p.DilithiumSigner
 	case Ed25519Type:
-		return &p.Ed25519Singer, nil
+		return &p.Ed25519Singer
 	default:
-		return nil, errUnknownSigner
+		panic(errUnknownSigner)
```

Comment:
The problem with panics is that it exposes your code to an attack, and you need to actively add a "recover" method to work around that. This is a risky, since it's not obvious to the caller that the underlying code might have that condition. On the flip side, returning an error is the safer way, since the caller knows that it should test all the returned error codes.